app/services/service_listeners/publishing_api_pusher.rb

```diff
@@ -39,6 +44,18 @@ def handle_html_attachments(event)
       PublishingApiHtmlAttachmentsWorker.perform_async(edition.id, event)
     end
 
+    # If the previous edition had extra translations, redirect them to the :en locale.
+    def handle_translations
+      removed_locales = edition.previous_edition.translations.map(&:locale) - edition.translations.map(&:locale)
```

Comment:
Could we just branch with `if edition.previous_edition` here and replace 34-38 could then just become `handle_translations` (I don't think we need the `begin->rescue` either, it will raise quite explicitly although I'm happy to merge with that in if you disagree). This will also mean you won't need the changes to `Edition`